File: AppSets-iOS/AppSets/com/app/appsets/worker/LastSyncWorker.swift
//
//  LastDataSyncWorker.swift
//  AppSets
//
//  Created by caiju Xu on 2024/7/12.
//

import Foundation

class LastSyncWorker : ListenableWorker {
    
    public static let TAG = "LastSyncWorker"
    
    public static let DATA_SYNC_FINISH = "data_sync_finish"
    
    override func doWork() async -> Result {
        PurpleLogger.current.d(LastSyncWorker.TAG, "doWork")
        LocalMessager.post(LastSyncWorker.DATA_SYNC_FINISH, nil)
        return Result.Success
    }
}

File: AppSets-Android/appsets/src/main/kotlin/xcj/app/appsets/worker/LastSyncWorker.kt
package xcj.app.appsets.worker

import android.content.Context
import android.content.Intent
import androidx.work.CoroutineWorker
import androidx.work.WorkerParameters
import xcj.app.appsets.service.MainService
import xcj.app.starter.android.util.LocalMessenger
import xcj.app.starter.android.util.PurpleLogger

class LastSyncWorker(
    context: Context,
    params: WorkerParameters
) : CoroutineWorker(context, params) {

    companion object {
        private const val TAG = "LastSyncWorker"
        const val MESSAGE_KEY_DATA_SYNC_FINISH = "data_sync_finish"
    }

    override suspend fun doWork(): Result {
        PurpleLogger.current.d(TAG, "doWork")
        LocalMessenger.post(MESSAGE_KEY_DATA_SYNC_FINISH, true)
        applicationContext.stopService(Intent(applicationContext, MainService::class.java))
        return Result.success()
    }
}